src/components/CompareResults/ResultsMain.tsx

```diff
@@ -91,18 +96,106 @@ function ResultsMain() {
     }
   }
 
+  /********** Edit Results Title Section **********/
+  const [rawSearchParams, updateRawSearchParams] = useRawSearchParams();
+  const initialComparisonTitle = rawSearchParams.get('title') ?? 'Results';
+  const [comparisonTitleName, setComparisonTitleName] = useState(
+    initialComparisonTitle,
+  );
+  const [previousComparisonTitle, setPreviousComparisonTitle] =
+    useState('Results');
+  const [editComparisonTitleInputVisible, showEditComparisonTitle] =
+    useState(false);
+
+  const handleEditInputToggle = () => {
+    if (editComparisonTitleInputVisible) {
+      setComparisonTitleName(previousComparisonTitle); //revert to previous title
+    } else {
+      setPreviousComparisonTitle(comparisonTitleName); //store current title as previous state
+    }
+
+    showEditComparisonTitle(!editComparisonTitleInputVisible);
+  };
+
+  const slugifyComparisonTitle = (title: string) => {
+    return (
+      title
+        /**
+         *please see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize#examples
+         */
+        .normalize('NFD') // Normalize to decompose diacritics (e.g., é -> e)
+        .replace(/[\u0300-\u036f]/g, '') // Remove accents
+        .toLowerCase()
+        .replace(/[^a-z0-9]+/g, '-') // Convert non-alphanumeric to hyphen
+        .replace(/^-+|-+$/g, '') // Trim hyphens from start and end
+    );
+  };
+
+  const onComparisonTitleChange = (value: string) => {
+    //PR notes: fix for truncated letter at the end of entry
+    const slug = slugifyComparisonTitle(value);
+    rawSearchParams.set('title', slug);
+    setComparisonTitleName(value);
+  };
+
+  const OnComparisonTitleSave = () => {
+    updateRawSearchParams(rawSearchParams);
+    showEditComparisonTitle(!editComparisonTitleInputVisible);
+  };
+
+  const deSlugify = (slug: string | undefined) => {
+    if (!slug) return '';
+    return slug.replace(/-/g, ' ').replace(/^\w/, (char) => char.toUpperCase());
+  };
+
+  const deSlugifiedTitle = deSlugify(comparisonTitleName);
+
+  const buttonIcon = (
+    <img
+      id='edit-title-icon'
+      className='icon icon-edit'
+      src={themeMode === 'light' ? pencil.toString() : pencilDark.toString()}
+      alt='edit-title-icon'
```

Comment:
nit: this alt is useless IMO
rather add `role="presentation"` for this presentational image